sa/storage-authority.go

```diff
@@ -166,6 +166,20 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 	return
 }
 
+// Get the valid authorization with biggest expire date for a given domain and registrationId
+func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64, identifier core.AcmeIdentifier) (authz core.Authorization, err error) {
+	ident, err := json.Marshal(identifier)
+	if err != nil {
+		return
+	}
+	err = ssa.dbMap.SelectOne(&authz, "SELECT id, identifier, registrationID, status, expires, challenges, combinations "+
+		"FROM authz "+
+		"WHERE identifier = :identifier AND registrationID = :registrationId "+
```

Comment:
fixed in https://github.com/r0ro/boulder/commit/236a90afdac3428ad3e10e03e7ba59e81f34dc96